scripts/symbolicate-profile.ts

```diff
@@ -0,0 +1,208 @@
+if (githubToken === undefined) {
+    Logger.error('No GitHub token provided. Either set a GITHUB_TOKEN environment variable or pass it using --gh-token');
+    process.exit(1);
+}
+// #endregion
+
+// #region Get the app version
+
+// Formatted as "Profile_trace_for_1.4.81-9.cpuprofile"
+const appVersionRegex = /\d+\.\d+\.\d+(-\d+)?/;
+const appVersion = argsMap.profile.match(appVersionRegex)?.[0];
+if (appVersion === undefined) {
+    Logger.error('Could not extract the app version from the profile filename.');
+    process.exit(1);
+}
+Logger.info(`Found app version ${appVersion} in the profile filename`);
+// #endregion
+
+// #region Utility functions
+// We need the token for the download step
+const octokit = new Octokit({auth: githubToken});
+const OWNER = 'Expensify';
+const REPO = 'App';
+
+function getWorkflowRunArtifact() {
+    const artefactName = `${argsMap.platform}-sourcemap-${appVersion}`;
+    Logger.info(`Fetching sourcemap artifact with name "${artefactName}"`);
+    return octokit
+        .request('GET /repos/{owner}/{repo}/actions/artifacts', {
+            owner: OWNER,
+            repo: REPO,
+            per_page: 1,
+            name: artefactName,
+        })
+        .then((artifactsResponse) => {
+            const artifact = artifactsResponse.data.artifacts[0];
+            if (artifact === undefined) {
+                throw new Error(`Could not find the artifact ${artefactName}!`);
+            }
+            return artifact.id;
+        })
+        .catch((error) => {
+            Logger.error('Failed to get artifact!');
+            Logger.error(error);
+            throw error;
+        });
+}
+
+function getDownloadUrl(artifactId: number) {
+    // https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#download-an-artifact
+    // Gets a redirect URL to download an archive for a repository. This URL expires after 1 minute.
+    // Look for Location: in the response header to find the URL for the download.
+
+    Logger.log(`Getting download URL for artifact…`);
+    return octokit
+        .request('GET /repos/{owner}/{repo}/actions/artifacts/{artifact_id}/{archive_format}', {
+            owner: OWNER,
+            repo: REPO,
+            artifact_id: artifactId,
+            archive_format: 'zip',
+        })
+        .then((response) => {
+            // The response should be a redirect to the actual download URL
+            const downloadUrl = response.url;
+
+            if (downloadUrl === undefined) {
+                throw new Error(`Could not find the download URL in:\n${JSON.stringify(response, null, 2)}`);
+            }
+
+            return downloadUrl;
+        })
+        .catch((error) => {
+            Logger.error('Failed to download artifact!');
+            Logger.error(error);
+            throw error;
+        });
+}
+
+const dirName = '.sourcemaps';
+const sourcemapDir = path.join(process.cwd(), dirName);
```

Comment:
NABish but I think this line will make it so this script only works from the root directory of App. Use this to make it slightly more robust:

```ts
const sourcemapDir = path.resolve(__dirname, '../.sourcemaps');
```

Reply:
excellent idea!